README.md

```diff
@@ -1,2 +1,941 @@
+
+Although, the creation of pages are a bit lengthy, **n-app** offers **separation of concern**, **ease of readability** and also **scalability**, if used correctly.
+
+<a id="page's-lifecycle-methods"></a>
+
+### **Page's Lifecycle Methods**
+
+**Note:** The following methods are inherited from the `PageViewModel` class.
+
+<a id="onCreate"></a>
+
+#### **onCreate**
+
+This method is executed when the **view-model** is created, but when the template is not yet mounted to the **DOM**.
+
+```typescript
+protected onCreate(): void
+{
+    super.onCreate();
+    
+    // Method Body
+}
+```
+
+<a id="onMount"></a>
+
+#### **onMount**
+
+This method is executed when the page template is mounted on the **DOM**, you'll get the HTML element as a parameter here to manipulate it, like using JQuery for example.
+
+```typescript
+protected onMount(element: HTMLElement): void
+{
+    super.onMount(element);
+    
+    // Method Body + Manipulation of the HTML element
+}
+```
+
+<a id="onDismount"></a>
+
+#### **onDismount**
+
+This method is executed when the page template is disMounted on the **DOM**.
+
+```typescript
+protected onDismount(): void
+{
+    super.onDismount();
+    
+    // Method Body
+}
+```
+
+<a id="onEnter"></a>
+
+#### **onEnter**
+
+This method is executed when the page has appeared and is usually used to fetch data to show on the page. The parameters for this function would be any **query/path params** of the url defined in the **route**.
+
+```typescript
+protected onEnter(): void
+{
+    super.onEnter();
+    
+    // Method Body
+}
+```
+
+<a id="onLeave"></a>
+
+#### **onLeave**
+
+This method is executed when the user is about to **leave** the page.
+
+```typescript
+protected onLeave(): void
+{
+    super.onLeave();
+    
+    // Method Body
+}
+```
+
+<a id="onDestroy"></a>
+
+#### **onDestroy**
+
+This method is executed when the page is **removed** from the **DOM**.
+
+```typescript
+protected onDestroy(): void
+{
+    super.onDestroy();
+    
+    // Method Body
+}
+```
+
+<a id="persisted-pages"></a>
+
+### **Persisted Pages**
+
+**n-app** allows you to have persisted pages which is means the state of a page is preserved throughout the whole session.
+
+Using the `@persist` decorator on top of the **view-model** will make the page persistence.
+
+```typescript
+@persist
+```
+
+**Note:** Persisted pages calls two lifecycle methods when entering a page and leaving a page, **onMount** and **onDismount**, respectively.
+
+**Note:** If a page is persisted, then **components** are also persisted.
+
+<a id="interaction-between-view-and-view-model"></a>
+
+### **Interaction between View and View-Model**
+
+**n-app** allows for Vue operations with the DOM with **separation of concern**. Instead of writing the methods, data and template inside one file, you are able to separate it into a template file (**HTML**), and a **view-model** file (**TypeScript**).
```

Comment:
This section should be at the very top, where you are explaining how to structure